### DISCUSSION TITLE
A separate Typescript compiler?

### DISCUSSION THREAD

**Discussion by nikitazdvijkov** [swc-project/swc#5337] [Q&A] [answered]
Hello,

Curious about SWC internals when it comes to Typescript...

SWC uses the regular official Typescript compiler under the hood, right?—Then how is the SWC compiler faster than `tsc`?

Related: SWC respects `tsconfig.json`, right?

Thanks.

**Comment by kwonoj** [comment, marked as answer]
> uses the regular official Typescript compiler under the hood, righ

No, this is not correct.

> SWC respects tsconfig.json, right

No.
